### PR TITLE
Ddp 5714 searchable dynamic form fields

### PR DESCRIPTION
**Pull request by Irakli-Khaladze**
Main changes are in FilterRoute and FilterTest. Other stuff is some refactorings.

### CODE REVIEW

File: src/main/java/org/broadinstitute/dsm/route/FilterRoute.java [hunk 1]
```diff
@@ -238,22 +244,30 @@ else if (savedFilters != null) {
 
     public static List<?> filterParticipantList(Filter[] filters, Map<String, DBElement> columnNameMap, @NonNull DDPInstance instance) {
         Map<String, String> queryConditions = new HashMap<>();
+        List<ParticipantDataDto> allParticipantData = participantDataDao
+                .getParticipantDataByInstanceid(Integer.parseInt(instance.getDdpInstanceId()));
```
Comment (SimoneMaiwald) [inline src/main/java/org/broadinstitute/dsm/route/FilterRoute.java]: I wonder if we really should have that here. If there is not filter for PARTICIPANT_DATA.equals(filter.getParticipantColumn().tableAlias) then we wouldn't need the list of participantData…

File: src/main/java/org/broadinstitute/dsm/route/PatchRoute.java [hunk 2]
```diff
@@ -91,7 +91,7 @@ public Object processRequest(Request request, Response response, String userId)
                                 if (patch.getActions() != null) {
                                     for (Value action : patch.getActions()) {
                                         if (ESObjectConstants.ELASTIC_EXPORT_WORKFLOWS.equals(action.getType())) {
-                                            writeESWorkflow(patch, nameValue, action);
+//                                            writeESWorkflow(patch, nameValue, action);
```
Comment (SimoneMaiwald) [inline src/main/java/org/broadinstitute/dsm/route/PatchRoute.java]: why is that commented out?

File: src/main/java/org/broadinstitute/dsm/route/PatchRoute.java [hunk 3]
```diff
@@ -255,7 +255,7 @@ else if (participantDataId != null) {
...
```
Comment (SimoneMaiwald) [inline src/main/java/org/broadinstitute/dsm/route/PatchRoute.java]: same here

File: src/main/java/org/broadinstitute/dsm/route/FilterRoute.java (hunk 1 above)
Comment (Irakli-Khaladze) [reply src/main/java/org/broadinstitute/dsm/route/FilterRoute.java]: Yes, at first I had it inside that if block you mentioned, but in this case we would need to do it for each filter with participantData and that's why moved it there. But I guess there won't be too many filters so maybe it was better to leave it inside the if block not to be called in every case. So do you think I should move it into the if block?

**Review by SimoneMaiwald — APPROVED** [review]

**Pull request merged**